### DISCUSSION TITLE
How to build a Community with cobra.Model instead of a sbml file path

### DISCUSSION THREAD

**Discussion by LeungYpH** [micom-dev/micom#86] [Q&A] [answered]
Hello!

Thanks for releasing **micom** and the tutorials.
I am trying to build a community with modified cobra models(adding new reactions and reset the bounds...)
but it seems that the Community can only be built with the dataFrame containing the file path of the sbml.
I try to do as the following
```
reaction1 = micom.util.Reaction('METt3pp__Ec1')
reaction1.name = 'L-methionine tranport out via proton antiport (periplasm) '
reaction1.subsystem = 'Cell Envelope Biosynthesis'
reaction1.lower_bound = 0.  # This is the default
reaction1.upper_bound = 1000.  # This is the default
reaction1.add_metabolites({
    com.metabolites.get_by_id('met__L_c__Ec1'): -1.0,
    com.metabolites.get_by_id('h_p__Ec1'): -1.0,

    com.metabolites.get_by_id('h_c__Ec1'): 1.0,
    com.metabolites.get_by_id('met__L_p__Ec1'): 1.0
})
com.add_reaction(reaction1)
```
but when optimizing, it will show the error
```
AttributeError: 'Reaction' object has no attribute 'community_id'
```
And I can not find the way to add the community_id and global_id for the reaction.

-----------------------------------------------------------------------------------------------------------------------
Actually, I can handle it by modifying the model and write it into sbml file and finally  read it into community.
But it will be better if this problem can be handled more elegantly.

Thanks!!!

**Comment by cdiener** [comment, marked as answer]
Yeah there is no good way to do that right now. For a single reaction, you can just add the community_id and global_id as you said. So for your code:

```python
reaction1 = micom.util.Reaction('METt3pp__Ec1')
reaction1.name = 'L-methionine tranport out via proton antiport (periplasm) '
reaction1.subsystem = 'Cell Envelope Biosynthesis'
reaction1.lower_bound = 0.  # This is the default
reaction1.upper_bound = 1000.  # This is the default
reaction1.add_metabolites({
    com.metabolites.get_by_id('met__L_c__Ec1'): -1.0,
    com.metabolites.get_by_id('h_p__Ec1'): -1.0,

    com.metabolites.get_by_id('h_c__Ec1'): 1.0,
    com.metabolites.get_by_id('met__L_p__Ec1'): 1.0
})
reaction1.community_id = "Ec1"
reaction1.global_id = "METt3pp"

com.add_reaction(reaction1)
```

I was considering allowing to have COBRAPY models in the "file" column of the taxonomy. Though, we were not sure about the use case, because if you want to publish the results you will probably need to provide the input models in a standardized format. So you would need them saved in a file. So modifying the original model and saving that to an SBML or JSON file before using it in MICOM sounds pretty clean to me. Though I am definitely open to have my mind changed on this.

**Reply by LeungYpH** [reply]
OK, thanks a lot !!!
